base/secretbuffer.jl

```diff
@@ -0,0 +1,153 @@
+"""
+    Base.SecretBuffer()
+
+An IOBuffer-like object where the contents will be securely wiped when garbage collected. However, it is
+considered best practice to wipe the buffer using `Base.shred!(::SecretBuffer)` as soon as the
+secure data are no longer required. Avoid initializing with converting to strings as they are
```

Comment:
"Avoid initializing with converting..." reads poorly. Perhaps first mention that they can be initialized, then add the note to avoid String second?

Reply:
Missed an "and" there —  I was intending: "Avoid initializing with and converting to".  Good point about swapping ordering here.